src/compat-filesystem.h

```diff
@@ -0,0 +1,17 @@
+#ifndef __COMPAT_FILESYSTEM__
+#define __COMPAT_FILESYSTEM__
+
+#if defined(__cplusplus) && __cplusplus >= 201703L && \
+    defined(__has_include) && __has_include(<filesystem>)
```

Comment:
GCC doc seems to say that checking for __has_include and using it on the same line is not recommended:
https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

Reply:
Done